Elk.py

```diff
@@ -0,0 +1,348 @@
+    except Exception as e:
+        print(e)
+
+
+def main():
+    tprint("ELK    EC2    SCAN")
+    link = input("insert your Elk URL (e.g: localhost:9200) : ")
+    username = input("insert your Elk username for auth(if there is no auth, click ENTER): ")
+    # (?) need to get the directories too (vuls and lynis), sudo password
+
+    if username != "":
+        password = input("insert your Elk password for auth: ")
+        elastic = Elasticsearch([link], http_auth=(username, password))
+    else:
+        elastic = Elasticsearch([link])
+    print("")
+    print("running ...")
+
+    # get IP:
+    hostname = socket.gethostname()
+    local_ip = socket.gethostbyname(hostname)
+
+    sudo_pass = ''  # sudo password of the machine
+    sudo_password = "echo " + sudo_pass + " | sudo -S "
+
+    begin_time = datetime.datetime.now()
+
+    vuls_directory = "home/ubuntu/idannos"
+    lynis_directory = "home/ubuntu/lynis"  # don't really use it the function, explanation there.
+
+    chkrotkit(sudo_password)
+    print("finished rootkit")
+
+    vuls(vuls_directory, sudo_password)
+    print("finished vuls")
+
+    lynis(lynis_directory, sudo_password)
+    print("finished lynis")
+
+    # get the public IP:
+    public_ip = str(get('https://api.ipify.org').text)
+
+    # get the instance_id:
+    instance_id = ec2_metadata.instance_id
+    # uuid:
+    uuid_string = str(uuid.uuid4())
+
+    # timeStamp:
+    date_and_hour = datetime.datetime.now()
+    temp = str(date_and_hour).split(" ")
+    date = temp[0]  # getting the date only without hours
+
+
+    # need to fill this before running:
+    try:
+        ACCESS_KEY = ''
+        SECRET_KEY = ''
+        iam = boto3.resource('iam',
+                             aws_access_key_id=ACCESS_KEY,
+                             aws_secret_access_key=SECRET_KEY,
+                             )
+        account_id = iam.CurrentUser().arn.split(':')[4]
+    except Exception as e:
+        account_id = "failed to get account_id"
+    # all of the uploading take about 8 seconds:
+    send_json_to_ELK("rootkit.json", "aws_rootkit_scan_", instance_id, date, account_id, uuid_string, "rootkit",elastic)
+    send_json_to_ELK("cves.json", "aws_vuls_cves_scan_", instance_id, date, account_id, uuid_string, "vuls", elastic)
+    send_json_to_ELK("lynis.json", "aws_lynis_scan_", instance_id, date, account_id, uuid_string, "lynis", elastic)
+    # how to see: in kibana -> settings -> index patterns -> create index pattern -> providing the names etc.
+
+    print("Took: ", datetime.datetime.now() - begin_time, " to execute.")
+    # locally about 2:30 minutes. less on ec2- about 1:30
+
+
+if __name__ == "__main__":
+    main()
+
+"""
+mini tutorial before running:
+- install Vuls with docker from: https://vuls.io/docs/en/tutorial-docker.html
+- install chkrootkit: apt-get install chkrootkit
+- install lynis: apt-get install lynis
+
+- Helping with setting auth to ELK: https://github.com/deviantony/docker-elk
+
```

Comment:
add a requirements.txt file to the repo.
[https://realpython.com/lessons/using-requirement-files/](url)

Reply:
link gives 404